### ISSUE TITLE
Reopen in Container stopped working after upgrade from 0.183.0 to 0.187.0

### ISSUE THREAD

**Issue by i-ky** [microsoft/vscode-remote-release#5310]
- VSCode Version: 1.58.0, 1.59.0-insider
- Local OS Version: Windows 10 / WSL 2
- Remote OS Version: Linux
- Remote Extension/Connection Type: Docker
- Logs (slightly redacted, I can't reveal all the details):
```
[2021-07-09T10:24:32.392Z] 
[+] Building 5.1s (6/6) FINISHED
 => [internal] load build definition from updateUID.Dockerfile-0.187.0     0.1s
 => => transferring dockerfile: 1.34kB                                     0.0s
 => [internal] load .dockerignore                                          0.2s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/library/foo_main:latest         0.0s
 => [1/2] FROM docker.io/library/foo_main                                  3.1s
 => [2/2] RUN eval $(sed -n "s/1000:1000:[^:]*:\([^:]*\):\([^:]*\):[^:]*:  1.0s
 => exporting to image                                                     0.6s
 => => exporting layers                                                    0.3s
 => => writing image sha256:653afc4fbdc5af2b2c3cca99ee0d6855d25e54c446442  0.0s
 => => naming to docker.io/library/vsc-foo-574bd3c779d16554147b1e56613544  0.0s
[2021-07-09T10:24:32.557Z] 
Use 'docker scan' to run Snyk tests against images to find vulnerabilities and l
earn how to fix them
[2021-07-09T10:24:32.658Z] Stop (5990 ms): Run: wsl -d Ubuntu-18.04 -e /bin/sh -c cd '/home/developer/FOO' && DISPLAY='192.168.1.78:0' ELECTRON_RUN_AS_NODE='1' SSH_ASKPASS='c:\Users\username\.vscode\extensions\ms-vscode-remote.remote-containers-0.187.0\scripts\ssh-askpass.bat' VSCODE_SSH_ASKPASS_NODE='C:\Users\username\AppData\Local\Programs\Microsoft VS Code\Code.exe' VSCODE_SSH_ASKPASS_MAIN='c:\Users\username\.vscode\extensions\ms-vscode-remote.remote-containers-0.187.0\dist\common\sshAskpass.js' VSCODE_SSH_ASKPASS_HANDLE='\\.\pipe\ssh-askpass-97185ce3bc2782b587e9bf4ffb312afc3849948f-sock' DOCKER_CONTEXT='default' VSCODE_SSH_ASKPASS_COUNTER='10' docker 'build' '-f' '/tmp/vsch/updateUID.Dockerfile-0.187.0' '-t' 'vsc-foo-574bd3c779d16554147b1e56613544e2-uid' '--build-arg' 'BASE_IMAGE=foo_main' '--build-arg' 'REMOTE_USER=1000:1000' '--build-arg' 'NEW_UID=1000' '--build-arg' 'NEW_GID=1000' '--build-arg' 'IMAGE_USER=1000:1000' '/tmp/vsch'
[2021-07-09T10:24:32.662Z] Error: Could not read file: undefined
	at Object.readFile (c:\Users\username\.vscode\extensions\ms-vscode-remote.remote-containers-0.187.0\dist\extension\extension.js:115:5818)
	at async Hte (c:\Users\username\.vscode\extensions\ms-vscode-remote.remote-containers-0.187.0\dist\extension\extension.js:250:6)
	at async YF (c:\Users\username\.vscode\extensions\ms-vscode-remote.remote-containers-0.187.0\dist\extension\extension.js:240:449)
	at async CZ (c:\Users\username\.vscode\extensions\ms-vscode-remote.remote-containers-0.187.0\dist\extension\extension.js:220:10180)
```

My `devcontainer.json` looks like this:
```
{
    "dockerComposeFile": [],
    "service": "main",
    "runServices": ["main", ...],
    "workspaceFolder": "/code"
}
```

Steps to Reproduce:
1. Open folder in VS Code (from WSL)
2. Click _Reopen in Container_

Does this issue occur when you try this locally?: N/A
Does this issue occur when you try this locally and all extensions are disabled?: Yes

Unfortunately, I cannot reproduce it with a minimal example, but I will be glad to provide more details of my setup if needed. It seems to be related to [updating UID and GID](https://github.com/microsoft/vscode-docs/blob/main/remote-release-notes/v1_58.md#update-uidgid-with-docker-compose-on-linux) added in 0.187.0 and probably is related to the empty `dockerComposeFile` array.

Disabling this new feature fixes the error:
```
    "updateRemoteUserUID": false,
```
Downgrading to 0.183.0 fixes it as well.

**Comment by wb-joona** [comment]
I just bumped into the exact same issue, and I'm also using `"dockerComposeFile": []` (so that we can use `docker-compose.override.yml` for per-developer overrides).

`"updateRemoteUserUID": false` fixes it for me as well.

**Comment by chrmarti** [comment]
Fixed in 0.188.0 (currently requires VS Code Insiders). Could someone seeing this verify the fix in that version? I'm considering to also fix it for VS Code 1.58.

**Comment by i-ky** [comment]
@chrmarti 0.188.0 seems completely broken for me. No matter what I set in `updateRemoteUserUID`, Docker Compose reports invalid configuration. I guess it sees only one of my compose files. (As I said initially, I have `"dockerComposeFile": []` in `devcontainer.json`. I rely on `COMPOSE_FILE=...` in `.env` file.)

**Comment by chrmarti** [comment]
@i-ky Thanks, could you give 0.188.1 a try?

**Comment by i-ky** [comment]
@chrmarti 0.188.1 does not work for me:
```
[4096 ms] Start: Run: wsl -d Ubuntu-18.04 -e /bin/sh -c cd '/home/developer/FOO' && <a bunch of variables here...> docker-compose '-f' '/path/to/workspace/path/to/docker-compose.yml' '-f' '/path/to/project/docker-compose.yml' 'config'
[4835 ms] 
[4835 ms] The WEB_SERVER_PORT variable is not set. Defaulting to a blank string.
The COMPOSE_PROJECT_NAME variable is not set. Defaulting to a blank string.
The Compose file '...' is invalid because:
services.web.ports contains an invalid type, it should be a number, or an object

[4835 ms] Exit code 1
```
Sorry, I probably should have mentioned that I have "non-root" compose file as primary one: `COMPOSE_FILE=path/to/docker-compose.yml:docker-compose.yml`. If same files are simply specified as `-f` options, Docker Compose will look for `.env` file in `path/to/`. Whereas if I just run `docker-compose` without `-f` options, it will look for `.env` file in workspace root.

If I comment out the service using `WEB_SERVER_PORT`, then it starts working. So if you wish, you can consider `updateRemoteUserUID` issue solved and I will report `.env` issue separately.

**Comment by chrmarti** [comment]
@i-ky Thanks, including a fix in 0.188.2, could you give that a try?

**Comment by i-ky** [comment]
@chrmarti, sure! 0.188.2 works nicely in my setup with all my configuration files reverted to 0.183.0 state. Thank you!

**Comment by Yehonal** [comment]
@chrmarti could you please release this fix for 1.58 as well, as soon as possible?

`overrideCommand` is very important for our use case but upgrading to 1.187 makes VSCode not working with `"dockerComposeFile": [],` 

Also setting `"updateRemoteUserUID": false` doesn't work for me. After rebuild the container I still have that error

**Comment by chrmarti** [comment]
Released for VS Code 1.58 with Remote-Containers 0.187.1.
